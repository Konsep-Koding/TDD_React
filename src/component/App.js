import React, { Component } from 'react'
import {Button} from 'react-bootstrap'
import { max_number } from '../helper'

 class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      gifts:[]
    };
  }

  addGift = () => {
    const { gifts } = this.state;

    gifts.push({ id: max_number(this.state.gifts.map(gift => gift.id))+1 });

    this.setState({ gifts });
  }

  render() {
    return (
    <div>
      <h2>Gift It</h2>
      <div className='gift-list'>
        {this.state.gifts.map(gift => (
          <div key={gift.id}></div>
        ))}
      </div>
      <Button className="btn-add" onClick={this.addGift}> Add Gift </Button>
    </div>
    )
  }
}

export default App

 