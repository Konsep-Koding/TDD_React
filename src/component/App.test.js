import React from 'react';
import { shallow, configure } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter(), disableLifecycleMethods: true });
const app = shallow(<App/>);


describe('App', () => {
  it('Render', () => {
    expect(app).toMatchSnapshot();
    });
     
    it('should showing the List', () => {
      expect(app.state().gifts).toEqual([]) 
    });

    describe('When click button `add-gift`', () => {
      beforeEach(() => {
      // Find By tags
        app.find('.btn-add').simulate('click')
      });

      afterEach(() =>{
        app.setState({ gifts: [] });
      })

      it('Add new  `New Gift On To State` Gift', () => {
        expect(app.state().gifts).toEqual([{id:1}])
      });
      
      it('Add simulate 1', () => {
        expect(app.find('.gift-list').children().length).toEqual(1)
      });
    });
    
 
});

