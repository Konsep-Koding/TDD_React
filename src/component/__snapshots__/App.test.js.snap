// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render 1`] = `ShallowWrapper {}`;
